### PR TITLE
All effects added to CLI

### PR DESCRIPTION
**Pull request by kbacha**
Implements the effects in the CLI. This actually uncovered a couple of
other things that needed to be refactored to work correctly.

First, `Flags` needed to be made public. In the process of exposing it I
renamed it to `Flags` from `Flag` since it contains multiple flags.

Second, there are three undocumented effects that needed to be
implemented. These include data_created, data_updated, and data_removed.

Lastly, the effects were added as a render to the cli.

### Is there a GIF that reflects how this work made you feel?
hrmph

### CODE REVIEW

File: cli/src/fmt/simple/effect.rs [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        append_to_buffer!(buf, "id:   {}", effect.id());
+        append_to_buffer!(buf, "kind: {}", effect.kind_name());
+        append_to_buffer!(buf, "details:");
+
+        Some(match *effect.kind() {
+            Kind::Account(ref kind) => account::render(buf, kind),
+            Kind::Signer(ref kind) => signer::render(buf, kind),
+            Kind::Trustline(ref kind) => trustline::render(buf, kind),
+            Kind::Trade(ref kind) => trade::render(buf, kind),
+            Kind::Data(ref kind) => data::render(buf, kind),
```
Comment (correlator) [inline cli/src/fmt/simple/effect.rs]: 👌

File: cli/src/ledgers.rs [hunk 2]
```diff
@@ -48,14 +48,14 @@ pub fn payments(client: &Client, matches: &ArgMatches) -> Result<()> {
     pager.paginate(iter, |result| match result {
         Ok(op) => {
             println!("ID:             {}", op.id());
-            match op.kind() {
-                &OperationKind::CreateAccount(ref create_account) => {
+            match *op.kind() {
```
Comment (correlator) [inline cli/src/ledgers.rs]: What's the rationale behind this change?

**Comment by robertDurst** [comment]
Amazing how many little doc details we have uncovered. Messaged a link to a google doc I created to keep track of these.

File: client/src/resources/effect/data.rs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+pub enum Kind {
+    /// Data was added to an account.
+    Created(Effect),
+    /// Data was removed from an account.
+    Removed(Effect),
+    /// Data was modified on an account.
+    Updated(Effect),
+}
+
+/// Contains details about the data that was chaged
```
Comment (correlator) [inline client/src/resources/effect/data.rs]: changed*

File: client/src/resources/effect/data.rs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+/// The type of change that was performed
+#[derive(Debug, Deserialize, Clone)]
+pub enum Kind {
+    /// Data was added to an account.
+    Created(Effect),
+    /// Data was removed from an account.
+    Removed(Effect),
+    /// Data was modified on an account.
+    Updated(Effect),
+}
```
Comment (correlator) [inline client/src/resources/effect/data.rs]: So this effect simply states that data was created, destroyed or updated, but no details beyond that.  Interesting 🤔

File: cli/src/ledgers.rs (hunk 2 above)
Comment (kbacha) [reply cli/src/ledgers.rs]: @correlator This is recommended by clippy. Instead of matching on a reference, they recommend deref'ing the enum.

File: client/src/resources/effect/data.rs (hunk 4 above)
Comment (kbacha) [reply client/src/resources/effect/data.rs]: As far as the data that I saw, yes.
